lib/Runtime/Library/LazyJSONString.h

```diff
@@ -0,0 +1,89 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+#pragma once
+
+namespace Js
+{
+
+struct JSONProperty;
+struct JSONArray;
+
+enum class JSONContentType : uint8
+{
+    Array,
+    Object,
+    Undefined,
+    Null,
+    True,
+    False,
+    Number,
+    String
+};
+
+typedef DListCounted<JSONProperty, Recycler> JSONObject;
+
+struct JSONNumberData
+{
+    Field(Var) value;
+    Field(JavascriptString*) string;
+};
+
+struct JSONProperty
+{
+    Field(JSONContentType) type;
+    Field(const PropertyRecord*) propertyRecord;
+    Field(JavascriptString*) propertyName;
+    union
+    {
+        Field(JSONNumberData) numericValue;
+        Field(JavascriptString*) stringValue;
+        Field(JSONObject*) obj;
+        Field(JSONArray*) arr;
+    };
+
+    JSONProperty()
```

Comment:
Looks like clang is complaining about a lack of a copy constructor for `JSONProperty` which is needed for its use in a DList

Reply:
Specifically because of const-ness: `candidate constructor (the implicit copy constructor) not viable: 1st argument ('const Js::JSONProperty') would lose const qualifier`